### PR TITLE
VCI-95 Refactoring

### PR DESCRIPTION
**Pull request by artem-dudarev**
https://virtocommerce.atlassian.net/browse/VCI-95

Most significant changes:
- Fixed formatting, spelling, naming, code style.
- Fixed all compilation warnings. No warnings are allowed from now on.
- Fixed potential NullReferenceExceptions and ArgumentNullExceptions.
- Fixed `Pack` target that didn't copy information from module manifest to the NuGet package.

I recommend checking each commit one by one to make it easier to understand what has changed.

### CODE REVIEW

**Comment by mvktsk** [comment]
Review task https://virtocommerce.atlassian.net/browse/VP-7576 has been created

File: GrabMigrator/Build.GrabMigrator.cs [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-         new GrabMigrator.GrabMigrator().Do(GrabMigratorConfig);
-     });
+    [Parameter("Grab-migrator config file")]
+    public static string GrabMigratorConfig { get; set; }
 
+    public Target GrabMigrator => _ => _
+        .Requires(() => GrabMigratorConfig)
+        .Executes(() =>
+        {
+            new GrabMigrator.GrabMigrator().Do(GrabMigratorConfig);
```
Comment (Vectorfield4) [inline GrabMigrator/Build.GrabMigrator.cs:12]: In usings:
`using GrabMigrator;`
```suggestion
            var grabMigrator = new GrabMigrator();
            grabMigrator.Do(GrabMigratorConfig);
```

File: GrabMigrator/GrabMigrator.cs [hunk 2]
```diff
@@ -8,60 +8,61 @@
...
 
 namespace GrabMigrator
 {
     /// <summary>
-    /// Grab-migrator target implementation
+    ///     Grab-migrator target implementation
     /// </summary>
     internal class GrabMigrator
     {
         public void Do(string configFilePath)
```
Comment (Vectorfield4) [inline GrabMigrator/GrabMigrator.cs:20]: `Do` method has huge cognitive complexity, so reading this code is a real pain. Suggest to use guard clause for `if (!string.IsNullOrEmpty(configFilePath))`  and `if (File.Exists(configFilePath))`, etc. Extract methods from `config?,Grab == true` and `if (config?.Apply == true)`;

File: PlatformTools/Build.PackageManager.cs [hunk 3]
```diff
@@ -3,130 +3,159 @@
...
+                    {
+                        if (new Version(existingModule.Version) > new Version(module.Version))
+                        {
+                            Logger.Error($"{module.Id}: Module downgrading isn't supported");
+                            continue;
+                        }
+
+                        Logger.Info($"Change version: {existingModule.Version} -> {module.Version}");
+                        existingModule.Version = module.Version;
+                    }
```
Comment (Vectorfield4) [inline PlatformTools/Build.PackageManager.cs:92]:
```suggestion
                    else if (new Version(existingModule.Version) > new Version(module.Version))
                    {
                        Logger.Error($"{module.Id}: Module downgrading isn't supported");
                    }
                    else
                    {
                        Logger.Info($"Change version: {existingModule.Version} -> {module.Version}");
                        existingModule.Version = module.Version;
                    }
```

**Review by Vectorfield4 — APPROVED** [review]

File: GrabMigrator/Build.GrabMigrator.cs (hunk 1 above)
Comment (artem-dudarev) [reply GrabMigrator/Build.GrabMigrator.cs:12]: It will lead to a compilation error:
error CS0118: 'GrabMigrator' is a namespace but is used like a type

**Review by krankenbro — APPROVED** [review]

**Pull request merged**